tests/backup/backup_test.go

```diff
@@ -60,14 +54,136 @@ func TearDownBackupRestore(bkpNamespaces []string, restoreNamespaces []string) {
+		dash.TestCaseBegin("Backup: BasicBackupCreateWithRules", "Creating backup with Rules", "", nil)
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+		}
+	})
+	It("Basic Backup Creation", func() {
+		Step("Validate applications and get their labels", func() {
+			ValidateApplications(contexts)
+			logrus.Infof("Create list of pod selector for the apps deployed")
+			for _, ctx := range appContexts {
+				for _, specObj := range ctx.App.SpecList {
+					if obj, ok := specObj.(*appsapi.Deployment); ok {
+						if contains(app_list, obj.Name) == true {
+							ps[obj.Name] = obj.Spec.Template.Labels
+						}
+					} else if obj, ok := specObj.(*appsapi.StatefulSet); ok {
+						if contains(app_list, obj.Name) == true {
+							ps[obj.Name] = obj.Spec.Template.Labels
+						}
+					}
+				}
+			}
+		})
+		Step("Creating rules for backup", func() {
+			dash.Info("Creating pre rule for deployed apps")
+			pre_rule_status := createRuleForBackup("backup-pre-rule", "default", app_list, "pre", ps)
+			dash.VerifyFatal(pre_rule_status, true, fmt.Sprintf("Verifying pre rule for backup"))
+			dash.Info("Creating post rule for deployed apps")
+			post_rule_status := createRuleForBackup("backup-post-rule", "default", app_list, "post", ps)
+			dash.VerifyFatal(post_rule_status, true, fmt.Sprintf("Verifying Post rule for backup"))
+		})
+		Step("Creating bucket,backup location and cloud setting", func() {
+			dash.Info("Creating bucket,backup location and cloud setting")
+			for _, provider := range providers {
+				bucketName := fmt.Sprintf("%s-%s", "bucket", provider)
+				CredName := fmt.Sprintf("%s-%s", "cred1", provider)
+				backup_location_name := fmt.Sprintf("%s-%s", "location1", provider)
+				CloudCredUID = uuid.New()
+				CloudCredUID_list = append(CloudCredUID_list, CloudCredUID)
+				BackupLocationUID = uuid.New()
+				CreateBucket(provider, bucketName)
+				CreateCloudCredential(provider, CredName, CloudCredUID, orgID)
+				time.Sleep(time.Minute * 1)
+				CreateBackupLocation(provider, backup_location_name, BackupLocationUID, CredName, CloudCredUID, bucketName, orgID)
+			}
+		})
+	})
+	JustAfterEach(func() {
+		dash.Info("Deleting the deployed apps after the testcase")
+		for i := 0; i < len(contexts); i++ {
+			opts := make(map[string]bool)
+			opts[SkipClusterScopedObjects] = true
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			err := Inst().S.Destroy(contexts[i], opts)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Verify destroying app %s, Err: %v", taskName, err))
+		}
+		dash.Info("Deleting backup rules created")
+		RuleEnumerateReq := &api.RuleEnumerateRequest{
+			OrgId: orgID,
+		}
+		ctx, err := backup.GetAdminCtxFromSecret()
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to fetch px-central-admin ctx: [%v]", err))
```

Comment:
use dash.VerifyFatal every where you have Expect

Reply:
Done!!